public/app/features/dashboard/components/ShareModal/ShareModal.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { PureComponent } from 'react';
+import { css } from 'emotion';
+import { GrafanaTheme } from '@grafana/data';
+import { stylesFactory, Modal, TabsBar, Tab, TabContent, withTheme, Themeable, Icon } from '@grafana/ui';
+import { DashboardModel, PanelModel } from 'app/features/dashboard/state';
+import { ShareLink } from './ShareLink';
+import { ShareSnapshot } from './ShareSnapshot';
+import { ShareExport } from './ShareExport';
+import { ShareEmbed } from './ShareEmbed';
+
+const getStyles = stylesFactory((theme: GrafanaTheme) => ({
+  modalHeaderTitle: css`
+    font-size: ${theme.typography.heading.h3};
+    padding-top: ${theme.spacing.sm};
+    margin: 0 ${theme.spacing.md};
+  `,
+  modalHeaderIcon: css`
+    margin-right: ${theme.spacing.md};
+    font-size: inherit;
+    &:before {
+      vertical-align: baseline;
+    }
+  `,
+}));
+
+const shareModalTabs = [
+  { label: 'Link', value: 'link' },
+  { label: 'Embed', value: 'embed' },
+  { label: 'Snapshot', value: 'snapshot' },
+  { label: 'Export', value: 'export' },
+];
+
+interface Props extends Themeable {
```

Comment:
for core components you can just get theme from config.theme instead, no need to use with withTheme HOC

Reply:
Ok, sounds good.